packages/compass-indexes/src/components/create-index-form/query-flow-section.tsx

```diff
@@ -166,14 +173,17 @@ const QueryFlowSection = ({
     });
   };
 
-  const handleQueryInputChange = useCallback((text: string) => {
-    setInputQuery(text);
-    setHasNewChanges(true);
-  }, []);
+  const handleQueryInputChange = useCallback(
+    (text: string) => {
+      setInputQuery(text);
+      setHasNewChanges(true);
+      onQueryUpdated();
```

Comment:
If you're moving has query updated to the store, all related state should also be moved to the store